strongbox-data-service/src/main/java/org/carlspring/strongbox/gremlin/dsl/EntityTraversalDsl.java

```diff
@@ -0,0 +1,162 @@
+package org.carlspring.strongbox.gremlin.dsl;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+
+import org.apache.tinkerpop.gremlin.process.traversal.Traversal;
+import org.apache.tinkerpop.gremlin.process.traversal.Traverser;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.GremlinDsl;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.structure.Edge;
+import org.apache.tinkerpop.gremlin.structure.Element;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+import org.apache.tinkerpop.gremlin.structure.VertexProperty.Cardinality;
+import org.carlspring.strongbox.data.domain.DomainObject;
+import org.carlspring.strongbox.data.domain.EntityHierarchyNode;
+import org.carlspring.strongbox.gremlin.adapters.UnfoldEntityTraversal;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * @author sbespalov
+ *
+ * @param <S>
+ * @param <E>
+ */
+@GremlinDsl(traversalSource = "org.carlspring.strongbox.gremlin.dsl.EntityTraversalSourceDsl")
+public interface EntityTraversalDsl<S, E> extends GraphTraversal.Admin<S, E>
```

Comment:
Could you please add javadoc to the methods in this interface so it becomes obvious what they are doing or are meant for?

Reply:
done, please review
feel free to ask if it's not clear